core/src/main/java/com/microsoft/applicationinsights/internal/perfcounter/PerformanceCounterContainer.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    /**
+     * Sets the timeout to wait between collection of Performance Counters.
+     *
+     * The number must be a positive number
+     *
+     * Note that the method will be effective if called before the first call to the 'register' method.
+     * @param collectingIntervalInMillis The timeout to wait between collection of Performance Counters.
+     */
+    public void setCollectingIntervalInMillis(long collectingIntervalInMillis) {
+        if (collectingIntervalInMillis <= 0) {
+            InternalLogger.INSTANCE.error("'%d' is an illegal value is ignored. Must be a positive number", collectingIntervalInMillis);
+            return;
+        }
+
+        this.collectingIntervalInMillis = collectingIntervalInMillis;
+    }
+
+    /**
+     * Gets the timeout in milliseconds that the container will wait between collections of Performance Counters.
+     * @return The timeout between collections.
+     */
+    public long getCollectingIntervalInMillis() {
+        return collectingIntervalInMillis;
+    }
+
+    /**
+     * Stopping the collection of performance data.
+     * @param timeout The timeout to wait for the stop to happen.
+     * @param timeUnit The time unit to use when waiting for the stop to happen.
+     */
+    public synchronized void stop(long timeout, TimeUnit timeUnit) {
+        if (!initialized) {
+            return;
+        }
+
+        ThreadPoolUtils.stop(threads, timeout, timeUnit);
+        initialized = false;
+    }
+
+    /**
+     * A private method that is called only when the container needs to start
+     * collecting performance counters data. The method will schedule a callback
+     * to be called, it will initialize a {@link com.microsoft.applicationinsights.TelemetryClient} that the Performance Counters
+     * will use to report their data, and it will also register itself a the {@link com.microsoft.applicationinsights.internal.shutdown.SDKShutdownActivity}
+     */
+    private void initialize() {
+        if (!initialized) {
+            synchronized (INSTANCE) {
+                if (!initialized) {
+                    createThreadToCollect();
+
+                    telemetryClient = new TelemetryClient();
+
+                    scheduleWork();
+
+                    initialized = true;
+                }
+            }
+        }
+    }
+
+    private void scheduleWork() {
+        threads.scheduleAtFixedRate(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        for (PerformanceCounter performanceCounter : performanceCounters.values()) {
+                            try {
+                                performanceCounter.report(telemetryClient);
+                            } catch (Throwable e) {
+                                InternalLogger.INSTANCE.error("Exception while reporting performance counter '%s': '%s'", performanceCounter.getId(), e.getMessage());
```

Comment:
Worth logging the counter type (i.e. simple name) since the id won't help you when looking at the logs.